src/Agent.Worker/JobRunner.cs

```diff
@@ -334,6 +334,25 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     return await CompleteJobAsync(jobServer, jobContext, message, TaskResult.Failed);
                 }
 
+                // check if knob DrainQueuesAfterTask has correct value
+                bool defaultDrainQueuesAfterTask = AgentKnobs.DrainQueuesAfterTask.GetValue<BuiltInDefaultKnobSource>(jobContext).AsBoolean();
+                try
+                {
+                    AgentKnobs.DrainQueuesAfterTask.GetValue(jobContext).AsBooleanStrict();
+                }
+                catch (Exception ex)
+                {
+                    switch (ex)
+                    {
+                        case ArgumentException _:
+                        case FormatException _:
+                            jobContext.Warning($"Knob 'AGENT_DRAIN_QUEUES_AFTER_TASK' has wrong value, default value '{defaultDrainQueuesAfterTask}' will be set");
```

Comment:
Would be better to use bool.TryParse pattern here to avoid the exception handling if possible.  Exception handling is more verbose and can make things difficult to debug (e..g if a different ArguemntException or FormatException is thrown than the expected ones

Reply:
We had a call and decided to revert this extra logic from my PR